internals/overlord/servstate/workloads_test.go

```diff
@@ -0,0 +1,373 @@
+// Copyright (c) 2025 Canonical Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package servstate_test
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "gopkg.in/check.v1"
+	"gopkg.in/yaml.v3"
+
+	"github.com/canonical/pebble/internals/overlord/servstate"
+	"github.com/canonical/pebble/internals/overlord/state"
+	"github.com/canonical/pebble/internals/plan"
+)
+
+var schemaTests = []struct {
+	summary         string
+	layers          []string
+	combinedSection *servstate.WorkloadsSection
+	combinedYAML    string
+	error           string
+}{{
+	summary:         "empty section",
+	combinedSection: &servstate.WorkloadsSection{},
+	combinedYAML:    `workloads: {}`,
+}, {
+	summary: "single null workload",
+	layers: []string{`
+workloads:
+    default:
+    `},
+	error: `workload "default" has a null value`,
+}, {
+	summary: "single workload with no override policy",
+	layers: []string{`
+workloads:
+    no-override-policy:
+        user-id: 1337
+    `},
+	error: `workload "no-override-policy" must define an "override" policy`,
+}, {
+	summary: "single workload with invalid override policy",
+	layers: []string{`
+workloads:
+    invalid-override-policy:
+        override: bazinga
+    `},
+	error: `workload "invalid-override-policy" has an invalid "override" policy: "bazinga"`,
+}, {
+	summary: "single workload with empty name",
+	layers: []string{`
+workloads:
+    "":
+        override: replace
+    `},
+	error: `workload "" cannot have an empty name`,
+}, {
+	summary: "single default workload",
+	layers: []string{`
+workloads:
+    default:
+        override: replace
+    `},
+	combinedSection: &servstate.WorkloadsSection{
+		Entries: map[string]*servstate.Workload{
+			"default": {
+				Name:     "default",
+				Override: plan.ReplaceOverride,
+			},
+		},
+	},
+	combinedYAML: `
+workloads:
+    default:
+        override: replace
+    `,
+}, {
+	summary: "single non-default workload",
+	layers: []string{`
+workloads:
+    default:
+        override: merge
+        environment:
+            foo: bar
+            bar: baz
+        user-id: 1337
+        user: alyssa
+        group-id: 1337
```

Comment:
Let's make the user, user-id and group, group-id different in these tests, to ensure the code isn't accidentally swapping the two.